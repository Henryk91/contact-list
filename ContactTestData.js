export function contactTestData() {
return [
    {
      "id": "wCQ9iHNbXqwLsYBz2QUM",
      "createdBy": "me",
      "name": "Bob Hope",
      "firstName" : "Bob",
      "lastName" : "Hope",
      "number" : ["07264841519", "51871544815"],
      "email" : ["wertytrew2q@gmail.com" , "wertytrew@gmail.com"],
      "dataLable" : [{"tag" : "email" ,"data" : "wertytrew2q@gmail.com"},{"tag" : "email" ,"data" : "wertytrew@gmail.com"}, {"tag" : "number" ,"data"  : "07264841519"} , {"tag" : "number" ,"data"  : "51871544815"}]
    },
    {
      "id": "wCQ9iHebXqwLaYBz2QUM",
      "createdBy": "me",
      "name": "Bobb Hope",
      "firstName" : "Bobb",
      "lastName" : "Hope",
      "number" : ["07264841519", "51871544815"],
      "email" : ["wertytrew2q@gmail.com" , "wertytrew@gmail.com"],
      "dataLable" : [{"tag" : "email" ,"data" : "wertytrew2q@gmail.com"},{"tag" : "email" ,"data" : "wertytrew@gmail.com"}, {"tag" : "number" ,"data"  : "07264841519"} , {"tag" : "number" ,"data"  : "51871544815"}]
    },
    {
      "id": "wCQ9iHNwXqwLaYBz2QUM",
      "createdBy": "me",
      "name": "Boby Hope",
      "firstName" : "Boby",
      "lastName" : "Hope",
      "number" : ["07264841519", "51871544815"],
      "email" : ["wertytrew2q@gmail.com" , "wertytrew@gmail.com"],
      "dataLable" : [{"tag" : "email" ,"data" : "wertytrew2q@gmail.com"},{"tag" : "email" ,"data" : "wertytrew@gmail.com"}, {"tag" : "number" ,"data"  : "07264841519"} , {"tag" : "number" ,"data"  : "51871544815"}]
    },
    {
      "id": "wCQ9iHNbXqxLaYBz2QUM",
      "createdBy": "me",
      "name": "Bobyy Hope",
      "firstName" : "Bobyy",
      "lastName" : "Hope",
      "number" : ["07264841519", "51871544815"],
      "email" : ["wertytrew2q@gmail.com" , "wertytrew@gmail.com"],
      "dataLable" : [{"tag" : "email" ,"data" : "wertytrew2q@gmail.com"},{"tag" : "email" ,"data" : "wertytrew@gmail.com"}, {"tag" : "number" ,"data"  : "07264841519"} , {"tag" : "number" ,"data"  : "51871544815"}]
    },
    {
      "id": "wCQ9qHNbXqwLaYBz2QUq",
      "createdBy": "me",
      "name": "Stevee Jobs",
      "firstName" : "Stevee",
      "lastName" : "Jobs",
      "number" : ["07264841519", "51871544815"],
      "email" : ["wertytrew2q@gmail.com" , "wertytrew@gmail.com"],
      "dataLable" : [{"tag" : "email" ,"data" : "wertytrew2q@gmail.com"},{"tag" : "email" ,"data" : "wertytrew@gmail.com"}, {"tag" : "number" ,"data"  : "07264841519"} , {"tag" : "number" ,"data"  : "51871544815"}]
    },
    {
      "id": "wCQ9iHNbXqwgaYBz2QUq",
      "createdBy": "me",
      "firstName" : "Steve",
      "lastName" : "Jobs",
      "name": "Steve Jobs",
      "number" : ["07264841519", "51871544815"],
      "email" : ["wertytrew2q@gmail.com" , "wertytrew@gmail.com"],
      "dataLable" : [{"tag" : "email" ,"data" : "wertytrew2q@gmail.com"},{"tag" : "email" ,"data" : "wertytrew@gmail.com"}, {"tag" : "number" ,"data"  : "07264841519"} , {"tag" : "number" ,"data"  : "51871544815"}]
    },
    {
      "id": "wCQ9gHNbXqwLaYBz2QUq",
      "createdBy": "me",
      "name": "Stteve Jobs",
      "firstName" : "Stteve",
      "lastName" : "Jobs",
      "number" : ["07264841519", "51871544815"],
      "email" : ["wertytrew2q@gmail.com" , "wertytrew@gmail.com"],
      "dataLable" : [{"tag" : "email" ,"data" : "wertytrew2q@gmail.com"},{"tag" : "email" ,"data" : "wertytrew@gmail.com"}, {"tag" : "number" ,"data"  : "07264841519"} , {"tag" : "number" ,"data"  : "51871544815"}]
    },
    {
      "id": "wCQ9iHNbXqwLaYBg2sdM",
      "createdBy": "me",
      "name": "Walt Sis",
      "firstName" : "Walt",
      "lastName" : "Sis",
      "number" : ["07264841519", "51871544815"],
      "email" : ["wertytrew2q@gmail.com" , "wertytrew@gmail.com"],
      "dataLable" : [{"tag" : "email" ,"data" : "wertytrew2q@gmail.com"},{"tag" : "email" ,"data" : "wertytrew@gmail.com"}, {"tag" : "number" ,"data"  : "07264841519"} , {"tag" : "number" ,"data"  : "51871544815"}]
    },
    {
      "id": "wCQ9igNqXqwLaYBz2sdM",
      "createdBy": "me",
      "name": "Waltt Sis",
      "firstName" : "Waltt",
      "lastName" : "Sis",
      "number" : ["07264841519", "51871544815"],
      "email" : ["wertytrew2q@gmail.com" , "wertytrew@gmail.com"],
      "dataLable" : [{"tag" : "email" ,"data" : "wertytrew2q@gmail.com"},{"tag" : "email" ,"data" : "wertytrew@gmail.com"}, {"tag" : "number" ,"data"  : "07264841519"} , {"tag" : "number" ,"data"  : "51871544815"}]
    },
    {
      "id": "wCQ9iHNbgqwLaYBz2sdM",
      "createdBy": "me",
      "name": "Walter Sis",
      "firstName" : "Walter",
      "lastName" : "Sis",
      "number" : ["07264841519", "51871544815"],
      "email" : ["wertytrew2q@gmail.com" , "wertytrew@gmail.com"],
      "dataLable" : [{"tag" : "email" ,"data" : "wertytrew2q@gmail.com"},{"tag" : "email" ,"data" : "wertytrew@gmail.com"}, {"tag" : "number" ,"data"  : "07264841519"} , {"tag" : "number" ,"data"  : "51871544815"}]
    },
    {
      "id": "wCQ9igNbXqwLaYBz2sdM",
      "createdBy": "me",
      "name": "Walty Sis",
      "firstName" : "Walty",
      "lastName" : "Sis",
      "number" : ["07264841519", "51871544815"],
      "email" : ["wertytrew2q@gmail.com" , "wertytrew@gmail.com"],
      "dataLable" : [{"email" : "wertytrew2q@gmail.com"},{"email" : "wertytrew@gmail.com"}, {"number" : "07264841519"} , {"number" : "51871544815"}]
    },
    ]
  }