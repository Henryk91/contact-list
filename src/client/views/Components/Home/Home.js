import React, { Component } from 'react';
import { Link } from 'react-router-dom';

export default class Home extends Component {
    constructor(props) {
        super(props);
        this.state = {
            contacts: null,
            list: null
        }
    }
    render() {
        let list = this.props.contacts
        return (
            <div id="home">
                {list ?
                    <div><br />{createList(list)}</div> :
                    <h3>Please add your name at the top then <br /> click My Contacts or All Contacts.</h3>
                }
            </div>
        );
    }
}
const createList = (contacts) => {
    let list = null
    if (contacts) {
        list = contacts.map((person) => {
            return (
                <div className="listNameButton" key={person.id}>
                    <Link style={{ textDecoration: 'none' }} to={`/contacts/${person.id}`}>
                        <h3>{person.firstName} {person.lastName}</h3>
                    </Link>
                </div>)
        })
    }
    return list
}
