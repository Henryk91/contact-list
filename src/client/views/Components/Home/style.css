/* .weekDays{
  display:grid;
  grid-template-columns:repeat(3, 1fr);
  grid-auto-rows: auto;
}
.weekDays :first-child{
  grid-column:1/4;
}
.weekDays :last-child{
  grid-column:2/3;
} */
.listNameButton{
  background: rgb(241, 241, 241);
  border: 1px solid rgb(185, 185, 185);
  margin-top: 2px;
  width: 80vw;
  margin-left: 10vw;
}

@media(min-width: 1000px){
  .listNameButton{
    width: 40vw;
    margin-left: 30vw;
  }
}