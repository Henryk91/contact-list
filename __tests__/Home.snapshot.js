
import Home from '../src/client/views/Components/Home/Home';
import renderer from 'react-test-renderer';
import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { contactTestData } from "../ContactTestData"

test('ContactDetail component renders all details', () => {
    const cData = contactTestData();
    const rendered = renderer.create(
        <Router>
            <Home contacts={cData} />
        </Router>
    );
    expect(rendered.toJSON()).toMatchSnapshot();
});