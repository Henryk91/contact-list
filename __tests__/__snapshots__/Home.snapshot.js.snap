// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactDetail component renders all details 1`] = `
<div
  id="home"
>
  <div>
    <br />
    <div
      className="listNameButton"
    >
      <a
        href="/contacts/wCQ9iHNbXqwLsYBz2QUM"
        onClick={[Function]}
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        <h3>
          Bob
           
          Hope
        </h3>
      </a>
    </div>
    <div
      className="listNameButton"
    >
      <a
        href="/contacts/wCQ9iHebXqwLaYBz2QUM"
        onClick={[Function]}
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        <h3>
          Bobb
           
          Hope
        </h3>
      </a>
    </div>
    <div
      className="listNameButton"
    >
      <a
        href="/contacts/wCQ9iHNwXqwLaYBz2QUM"
        onClick={[Function]}
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        <h3>
          Boby
           
          Hope
        </h3>
      </a>
    </div>
    <div
      className="listNameButton"
    >
      <a
        href="/contacts/wCQ9iHNbXqxLaYBz2QUM"
        onClick={[Function]}
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        <h3>
          Bobyy
           
          Hope
        </h3>
      </a>
    </div>
    <div
      className="listNameButton"
    >
      <a
        href="/contacts/wCQ9qHNbXqwLaYBz2QUq"
        onClick={[Function]}
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        <h3>
          Stevee
           
          Jobs
        </h3>
      </a>
    </div>
    <div
      className="listNameButton"
    >
      <a
        href="/contacts/wCQ9iHNbXqwgaYBz2QUq"
        onClick={[Function]}
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        <h3>
          Steve
           
          Jobs
        </h3>
      </a>
    </div>
    <div
      className="listNameButton"
    >
      <a
        href="/contacts/wCQ9gHNbXqwLaYBz2QUq"
        onClick={[Function]}
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        <h3>
          Stteve
           
          Jobs
        </h3>
      </a>
    </div>
    <div
      className="listNameButton"
    >
      <a
        href="/contacts/wCQ9iHNbXqwLaYBg2sdM"
        onClick={[Function]}
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        <h3>
          Walt
           
          Sis
        </h3>
      </a>
    </div>
    <div
      className="listNameButton"
    >
      <a
        href="/contacts/wCQ9igNqXqwLaYBz2sdM"
        onClick={[Function]}
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        <h3>
          Waltt
           
          Sis
        </h3>
      </a>
    </div>
    <div
      className="listNameButton"
    >
      <a
        href="/contacts/wCQ9iHNbgqwLaYBz2sdM"
        onClick={[Function]}
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        <h3>
          Walter
           
          Sis
        </h3>
      </a>
    </div>
    <div
      className="listNameButton"
    >
      <a
        href="/contacts/wCQ9igNbXqwLaYBz2sdM"
        onClick={[Function]}
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        <h3>
          Walty
           
          Sis
        </h3>
      </a>
    </div>
  </div>
</div>
`;
